repo/db/cafe_sessions.go

```diff
@@ -92,17 +94,24 @@ func (c *CafeSessionDB) handleQuery(stm string) []repo.CafeSession {
 			continue
 		}
 
-		var rcafe repo.Cafe
+		var rcafe *pb.Cafe
 		if err := json.Unmarshal(cafe, &rcafe); err != nil {
 			log.Errorf("error unmarshaling cafe: %s", err)
 			continue
 		}
 
-		ret = append(ret, repo.CafeSession{
+		time := time.Unix(0, expiryInt)
+
+		timestamp := timestamp.Timestamp{
+			Seconds: time.Unix(),
+			Nanos:   int32(time.Nanosecond()),
+		}
```

Comment:
Pulling nanoseconds out of the db as `expiryInt`, creating a golang `Time` from it, then using that to create a proto `Timestamp`. Look right @sanderpick ?

Reply:
Ah yea much cleaner, updated in both places.